lang/attribute/account/src/lib.rs

```diff
@@ -66,29 +67,36 @@ pub fn account(
 ) -> proc_macro::TokenStream {
     let mut namespace = "".to_string();
     let mut is_zero_copy = false;
+    let mut skip_init_space = false;
```

Comment:
What's the logic behind having a `skip_init_space` option? 

Is it to reduce the code that goes into the compiled `.so` file? Is then
- I don't think there's any need for this option, we can just add `InitSpace` to `#[derive(AnchorSerialize, AnchorDeserialize, Clone)]` by default. 
- The logic being that if the `::InitSpace` constant is never called then the implementation never gets added to the compiled `.so`. The solana bpf compiler is pretty smart at leaving stuff out that doesn't get touched downstream of the `entrypoint` function.

Reply:
To avoid some conflicts. For example the IDL stuff in the lib

Reply:
Can you be more specific about the conflicts? I'm not sure understand how having a trait constant `InitSpace` could cause a conflict in the IDL?

Reply:
Oh, you mean how you had to add `skip_space` to the IDL account. I'll have to look into why that was causing a conflict?

Reply:
Ok, I see now. Your new macro won't work unless someone has specified the `max_len` value over the vecs. I'll have to think on this.

Reply:
Cool, i didn't realise there would be backwards compatibility issues even if the `INIT_SPACE` constant was never used. In that case, I'm not fully comfortable with adding it by default to the account attribute macro (sorry!)

The two possible solutions are
- don't add `InitSpace` to the account attribute, instead just export the InitSpace derive macro and make sure that it's well documented if anyone wants to manually add it on top of their account structs. (they should because it will be very handy to have!)
- A backwards compatible solution could be: instead of returning a compile error when the parsing code detects a missing `#[max_len()]` above string or vec, or any of the other parsing error, instead add a `panic!("")` with the appropriate error message. This would allow it to be backwards compatible, since the panic wouldn't trigger a compile error unless the variable was actually used. Though I doubt this approach would be best practice:
```
impl Space for IdlAccount {
    const INIT_SPACE: usize = {
        panic!("The INIT_SPACE trait requires the max_len attribute to be specified for vec types");
    };
}
```

Reply:
Thanks!